src/components/popover/types.ts

```diff
@@ -0,0 +1,82 @@
+import { Dispatch, MutableRefObject, ReactNode, SetStateAction } from 'react'
+
+export interface DialogArrowProps {
+  /**
+   * Whether the arrow should be shown.
+   */
+  shown: boolean
+
+  /**
+   * DOMRect of the target element, for positioning the arrow.
+   */
+  triggerRect: DOMRect
+
+  /**
+   * Minimum distance in pixels that the arrow should be from the viewport edge.
+   */
+  collisionBuffer: number
+
+  /**
+   * Size in pixels of the arrow.
+   */
+  arrowSize: number
+
+  /**
+   * Arrow coloration.
+   */
+  themeClass: string
+
+  /**
+   * An object of style properties.
+   */
+  themeProps: $TSFixMe
```

Comment:
I think this can be updated to `CSSProperties`? Looks like it's being spread into an object being cast as that type.

Reply:
Great catch! Updated in https://github.com/hashicorp/dev-portal/pull/266/commits/582226575965eddebd117c0bf08894bf20cb9fd0

Reply:
Do you have an opinion on module augmentation like described here? If I understand it correctly, it'll prevent us from having to cast so many objects in this component `as CSSProperties`.

https://github.com/frenic/csstype#what-should-i-do-when-i-get-type-errors

Reply:
I think that's an awesome idea! I think the only augmentation I'd like to see would be adding `[index: `--${string}`]: any;` to support custom properties. Adding `[index: string]: any;` would be a bit too permissive.

Reply:
Nice! That's a great way to limit it to CSS variables, our whole goal! I'll write up a ticket for it. 😊

Reply:
New ticket here: https://app.asana.com/0/1201987349274776/1202090858665463/f